zio-http/src/main/scala/zio/http/ZClient.scala

```diff
@@ -562,133 +556,166 @@ object ZClient {
         ).withFinalizer(_.close.orDie)
       } yield res
 
-    private def requestAsync(request: Request, clientConfig: ClientConfig)(implicit
-      trace: Trace,
-    ): ZIO[Any, Throwable, Response] = {
+    private def requestAsync(request: Request, clientConfig: ClientConfig): ZIO[Any, Throwable, Response] =
```

Comment:
I think just because we have merged changes on this a bunch of times and it compiles. The bigger issue is why this compiles even though there is no trace. I think the problem is we still are defining an empty trace in the package object and that needs to get deleted.